src/util.h

```diff
@@ -49,6 +50,20 @@ NORETURN void Fatal(const char* msg, ...);
 #define NINJA_FALLTHROUGH
 #endif
 
+/// Whether C++11 features are enabled or not.
+#if __cplusplus >= 201103L || (defined(_MSC_VER) && _MSC_VER >= 1900)
+#  define NINJA_CPP11 true
```

Comment:
Minor, newer versions of the standard have changed from just doing monolithic version checks to instead doing #defines based on individual features being supported by that version of the compiler / standard lib.

So since in this situation, you're adding this new definition *specifically* to check if there support for std::unique_ptr, you might consider changing this from NINJA_CPP11, to NINJA_HAS_UNIQUE_PTR, or something of that nature.

That being said, I don't see a problem with what you have here. It's just a minor recommendation.

Reply:
@jonesmz 

Umm... I had found many C++11 checks in different ways, so I wanted to unify them using `NINJA_CPP11` in the next PR.

https://github.com/ninja-build/ninja/search?q=%23if+__cplusplus+%3E%3D+201103L

WDYT?

Reply:
Yea, there's a lot of dispersed CPP11 checks, i agree.

I think my breakdown of preferences, in order, are

1. If your compiler / platform doesn't support C++11, you're using something very broken -- aka, mandatory C++11 minimum
2. Per feature detection. E.g. `std::make_unique` has the feature detection macro: [__cpp_lib_make_unique](https://en.cppreference.com/w/cpp/utility/feature_test) -- I think starting with C++14, basically every feature has a detection macro. If Ninja isn't going to fully adopt C++11, then my personal preference would be to adopt our own C++11 feature detection macros to make more fine-grained support possible for the various broken compilers / platforms out there.
3. Standardize all C++11 features under NINJA_CPP11 like you have here.